File: 2021/1/solutions/node/stelar7/main.mjs
import { readFileSync } from "fs";

const STANDARD_IN = 0
const lines = readFileSync(STANDARD_IN).toString().split("\n").slice(0, -1)

let prev = null;
let count = 0;
for (const line of lines) {
  const current = parseInt(line);
  if (prev != null && current > prev) count++;
  prev = current;
}

console.log(`${count}`);

count = 0;
prev = null;
for (let i = 0; i<lines.length; i++) {
  const current = parseInt(lines[i]) + parseInt(lines[i + 1]) + parseInt(lines[i + 2]);

  if (prev != null && current > prev) {
    count++;
  }
  prev = current;
}

console.log(`${count}`);

File: 2021/1/solutions/java/stelar7/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = null;

        List<Integer> data = new ArrayList<Integer>();
        while ((line = br.readLine()) != null) {
            data.add(Integer.parseInt(line));
        }

        int current = 0;
        int last = 0;
        int count = 0;
        int current2 = 0;
        int last2 = 0;
        int count2 = 0;

        for (int i = 0; i < data.size(); i++) {
            current = data.get(i);
            
            if (i < data.size() - 2) {
                current2 = data.get(i) + data.get(i + 1) + data.get(i + 2);
            }

            if (i > 0) {
                if (current > last) {
                    count++;
                }

                if (current2 > last2) {
                    count2++;
                }
            }

            last = current;
            last2 = current2;
        }

        System.out.println(count);
        System.out.println(count2);
    }
}